# Ⅰ
写一个函数，输入 n ，求斐波那契（Fibonacci）数列的第 n 项（即 F(N)）。斐波那契数列的定义如下：
```
F(0) = 0,   F(1) = 1
F(N) = F(N - 1) + F(N - 2), 其中 N > 1.
```
斐波那契数列由 0 和 1 开始，之后的斐波那契数就是由之前的两数相加而得出。

答案需要取模 1e9+7（1000000007），如计算初始结果为：1000000008，请返回 1。

来源：力扣（LeetCode）
链接：https://leetcode.cn/problems/fei-bo-na-qi-shu-lie-lcof
著作权归领扣网络所有。商业转载请联系官方授权，非商业转载请注明出处。

In [ ]:
#分治的思路
def fib(n: int) -> int:
    if n==1:
        return 0
    if n==2:
        return 1
    else:
        return fib(n-1)+fib(n-2)

In [10]:
#一维动态规划

def fib(n: int) -> int:
    if n==0:
        return 0
    if n==1:
        return 1
    f1=0
    f2=1
    ans=0
    for _ in range(1,n):
        ans=(f1+f2)%(1e9+7)
        f1=f2
        f2=ans
    return ans
fib(81)

107920472.0

注意这道题的坑：是对每一个值进行求余，而不是最终结果求余。

如果是最终结果求余，那么fib(81)=107920470。如果是每一个结果求余，结果是fib(81)=107920472

注意结果取整

# Ⅱ
一只青蛙一次可以跳上1级台阶，也可以跳上2级台阶。求该青蛙跳上一个 n 级的台阶总共有多少种跳法。

答案需要取模 1e9+7（1000000007），如计算初始结果为：1000000008，请返回 1。

来源：力扣（LeetCode）
链接：https://leetcode.cn/problems/qing-wa-tiao-tai-jie-wen-ti-lcof
著作权归领扣网络所有。商业转载请联系官方授权，非商业转载请注明出处。

程序和Ⅰ一摸一样，就是初始值进行更改。
思路：跳到第n个台阶，有两种方式：第一种，从第n-1个台阶跳（只跳一个），或者从第n-2个台阶跳（跳两个）。因此得出递推式$f(n)=f(n-1)+f(n-2)$。初始条件：f(0)=1,f(1)=1,f(2)=2